cmd/snapshotgc/snapshotgc.go

```diff
@@ -29,6 +36,66 @@ type snapshotData struct {
 	release            releasev1alpha1.Release
 }
 
+// Iterates tenant namespaces and garbage-collect their snapshots
+func garbageCollectSnapshots(
+	cl client.Client,
+	logger logr.Logger,
+	prSnapshotsToKeep, nonPrSnapshotsToKeep int,
+) error {
+	req, _ := labels.NewRequirement(
+		"toolchain.dev.openshift.com/type", selection.In, []string{"tenant"},
+	)
+	selector := labels.NewSelector().Add(*req)
+	namespaceList := &core.NamespaceList{}
+	err := cl.List(
+		context.Background(),
+		namespaceList,
+		&client.ListOptions{LabelSelector: selector},
+	)
+	if err != nil {
+		logger.Error(err, "Failed listing namespaces")
+		return err
+	}
+
+	logger.V(1).Info("Snapshot garbage collection started...")
+	for _, ns := range namespaceList.Items {
+		logger.V(1).Info("Processing namespace", "namespace", ns.Name)
+
+		snapToData := make(map[string]snapshotData)
+		snapToData, err = getSnapshotsForNSReleases(cl, snapToData, ns.Name, logger)
+		if err != nil {
+			logger.Error(
+				err, "Failed getting release snapshots. Skipping namespace "+ns.Name)
```

Comment:
This will allow us better processing of JSON logs
```suggestion
				err, "Failed getting release snapshots. Skipping namespace", "namespace", ns.Name)
```

Reply:
done